### PR TITLE
Yaml tables for ensembles and nests

### LINKED ISSUE #1568: Add ability to input different forcing files for each ensemble member in ensemble manager
**Is your feature request related to a problem? Please describe.**
The ensemble manager currently expects the forcing files for each ensemble member to be the same, so you cannot use it to spawn ensembles of predictions that are forced by different data sources.

**Describe the solution you'd like**
The ability to specify a unique forcing for each ensemble member in the ensemble manager. 

**Describe alternatives you've considered**
You can specify each ensemble member as a unique experiment and it's forcing data as unique data sources within the xml, but this can become quite long and difficult to edit for experiments with many ensemble members. 

**Additional context**

### LINKED ISSUE #1573: Configurable Diag_table Output Frequency Across Multiple Grids
It is desirable to configure the Diag_table to support different output frequencies for various grids within the same component. For instance, in a global-nest configuration, users are typically more interested in high-frequency output from the nested grid, while the global grid can be configured with a lower frequency to save disk space and simulation time.

### PR DESCRIPTION
**Pull request by uramirez8707**
**Description**
Adds the ability for each ensemble/nest to have its own diag/field/data yaml tables.

This will allow users to specify a unique forcing file for each ensemble member  (Fixes #1568) and to specify a different file frequency for multiple grids (Fixes #1573)

For example, for a run with 2 ensemble members, fms will search for diag_table_ens_01.yaml and diag_table_ens_02.yaml. 
Similarly, for a run with a nest, fms will search for diag_table.yaml (for the parent grid) and diag_table_nest_01.yaml (for the nest grid), `nest_01` can be whatever is passed in to fms via the `set_filename_appendix` call.

Fixes # (issue)

**How Has This Been Tested?**
CI including new tests

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

File: data_override/include/data_override.inc [hunk 1]
```diff
@@ -591,9 +591,18 @@ subroutine read_table_yaml(data_table)
     integer :: nentries, mentries
     integer :: i
     character(len=50) :: buffer
+    character(len=255) :: filename !< Name of the expected data_table.yaml
```
Comment (bensonr) [inline data_override/include/data_override.inc:594]: This should use the filename length macro in platform.F90

File: data_override/include/data_override.inc [hunk 2]
```diff
@@ -591,9 +591,18 @@ subroutine read_table_yaml(data_table)
...
-    file_id = open_and_parse_file("data_table.yaml")
+    ! If doing and ensemble or nest run add the filename appendix (ens_XX or nest_XX) to the filename
+    call get_instance_filename("data_table.yaml", filename)
+    if (index(trim(filename), "ens_") .ne. 0) then
+      if (file_exists(filename) .and. file_exists("data_table.yaml")) &
+        call mpp_error(FATAL, "Both data_table.yaml and "//trim(filename)//" exists, pick one!")
+    endif
+
+    file_id = open_and_parse_file(trim(filename))
+
```
Comment (bensonr) [inline data_override/include/data_override.inc:605]: Need to make it clear in any documentation we are only testing for collisions with `ens_xx` and a generic data_table.yaml and not for `nest_xx` as well.

File: diag_manager/fms_diag_yaml.F90 [hunk 3]
```diff
@@ -381,10 +382,17 @@ subroutine diag_yaml_object_init(diag_subset_output)
                                            !! outputing data at every frequency)
   character(len=:), allocatable :: filename!< Diag file name (for error messages)
   logical              :: is_instantaneous !< .True. if the file is instantaneous (i.e no averaging)
+  character(len=255)   :: yamlfilename     !< Name of the expected diag_table.yaml
```
Comment (bensonr) [inline diag_manager/fms_diag_yaml.F90:385]: This should use the filename length macro in platform.F90

File: diag_manager/fms_diag_yaml.F90 [hunk 4]
```diff
@@ -381,10 +382,17 @@ subroutine diag_yaml_object_init(diag_subset_output)
...
   if (diag_yaml_module_initialized) return
 
-  diag_yaml_id = open_and_parse_file("diag_table.yaml")
+  ! If doing and ensemble or nest run add the filename appendix (ens_XX or nest_XX) to the filename
+  call get_instance_filename("diag_table.yaml", yamlfilename)
+  if (index(trim(yamlfilename), "ens_") .ne. 0) then
+    if (file_exists(yamlfilename) .and. file_exists("diag_table.yaml")) &
+      call mpp_error(FATAL, "Both diag_table.yaml and "//trim(yamlfilename)//" exists, pick one!")
+  endif
+  diag_yaml_id = open_and_parse_file(trim(yamlfilename))
```
Comment (bensonr) [inline diag_manager/fms_diag_yaml.F90:395]: Need to make it clear in any documentation we are only testing for collisions with ens_xx and a generic data_table.yaml and not for nest_xx as well.

File: field_manager/field_manager.F90 [hunk 5]
```diff
@@ -606,18 +606,27 @@ subroutine read_field_table_yaml(nfields, table_name)
 logical                         :: fm_success !< logical for whether fm_change_list was a success
 logical                         :: subparams !< logical whether subparams exist in this iteration
 
+character(len=255) :: filename !< Name of the expected field_table.yaml
+
```
Comment (bensonr) [inline field_manager/field_manager.F90:610]: filename macro

File: field_manager/field_manager.F90 [hunk 6]
```diff
@@ -606,18 +606,27 @@ subroutine read_field_table_yaml(nfields, table_name)
...
 endif
-if (.not. file_exists(trim(tbl_name))) then
+
+call get_instance_filename(tbl_name, filename)
+if (index(trim(filename), "ens_") .ne. 0) then
+  if (file_exists(filename) .and. file_exists(tbl_name)) &
+    call mpp_error(FATAL, "Both "//trim(tbl_name)//" and "//trim(filename)//" exists, pick one!")
+endif
+
+if (.not. file_exists(trim(filename))) then
```
Comment (bensonr) [inline field_manager/field_manager.F90:623]: Same message about checking only `ens_xx`

**Review by bensonr — CHANGES_REQUESTED** [review]

**Comment by uramirez8707** [comment]
@bensonr I think I addressed all your comments in https://github.com/NOAA-GFDL/FMS/pull/1585/commits/1b2e050c9096eb3399996f8c276f45011385ac24

**Review by bensonr — DISMISSED** [review]
Thanks for making the changes.

**Review by rem1776 — APPROVED** [review]

**Pull request merged**